### PR TITLE
Feature: added support for video text tracks

### PR DESCRIPTION
**Pull request by fredguile**
(actually, as "pre-work" from my side, you will find two other changes:
- some prettier reformatting based on the provided `.prettierrc.json`
- I've refactored the example page putting the playback speed control in the controls at the bottom, and making sure that the video fits the screen. Simply because I couldn't see the subtitles correctly at first.

I had to "refresh" the example page to make it looks suitable for subtitles:
Screen Shot 2021-04-11 at 7 26 49 pm

Now regarding the video text tracks, it's all based on this HTML5 standard: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/track

The two additions are:
- passing a `textTracks` prop to list all available text tracks (subtitles, captions, metadata, ...) for all languages,
- `VideoState` gets a new property for inspecting the current active tracks

All text tracks are using this web standard: https://www.w3.org/TR/webvtt1/#introduction

Subtitles/captions can be rendered natively by the browser, but I've left the possibility for users to hide the default rendering (see our example page) and create their own React elements using the `VideoState.currentActiveCues()` function. This is particularly import for "metadata" tracks that don't render natively, and where users are asked to implement the rendering of these tracks themselves.

So with non-native english subtitles, our example  page now looks like this:

Screen Shot 2021-04-11 at 7 33 21 pm

### CODE REVIEW

File: src/video.tsx [hunk 1]
```diff
@@ -15,6 +9,7 @@ export type VideoError = MediaError | null;
 export interface VideoState {
   status: VideoStatus;
   currentTime: number;
+  currentActiveCues: (kind: VideoTextTrackKind, lang: string) => TextTrackCueList | null | undefined;
```
Comment (zzarcon) [inline src/video.tsx:12]: I think this goes against the current design, up to this point everything under `videoState` was plain properties and now we got a method. 

Isn't it the `actions` a more suitable place for this?

File: src/index.ts [hunk 2]
```diff
@@ -1,2 +1,15 @@
...
+  VideoStatus,
+  VideoError,
+  VideoState,
+  NavigateFunction,
+  SetVolumeFunction,
+  SetPlaybackSpeed,
+  VideoActions,
+  RenderCallback,
+  SourceElement,
+} from './video';
```
Comment (zzarcon) [inline src/index.ts:13]: Nice!

File: src/video.tsx [hunk 3]
```diff
@@ -25,7 +20,7 @@ export interface VideoState {
 
 export type NavigateFunction = (time: number) => void;
 export type SetVolumeFunction = (volume: number) => void;
-export type SetPlaybackSpeed = (volume: number) => void;
+export type SetPlaybackSpeed = (speed: number) => void;
```
Comment (zzarcon) [inline src/video.tsx:23]: LOL! nice one

File: README.md [hunk 4]
```diff
@@ -56,21 +57,23 @@ interface Props {
...
+  textTracks?: VideoTextTracks;
 }
 ```
 
-**render method**
+### Render method
 
 ```typescript
-type RenderCallback = (videoElement: ReactNode, state: VideoState, actions: VideoActions) => ReactNode;
+type RenderCallback = (reactElement: ReactElement<HTMLMediaElement>, state: VideoState, actions: VideoActions, ref: React.RefObject<HTMLMediaElement>) => ReactNode;
```
Comment (zzarcon) [inline README.md:67]: For a second I thought we never had a `ref`... and nothing made sense, later I realized docs where not sync! 
Thanks!!

File: README.md [hunk 5]
```diff
@@ -140,6 +143,46 @@ interface VideoActions {
 </Video>
 ```
 
-# Author 🧔
+## Video text tracks 🚂
+
+> HTML5 [text tracks](https://html.spec.whatwg.org/multipage/media.html#the-track-element) support for videos.
+>
+> subtitles can be rendered natively, or they can be rendered using `VideoState.currentActiveCues` property:
```
Comment (zzarcon) [inline README.md:150]: Is the `currentActiveCues` something that we need right now? As in, the existing use case requires a custom renderer? 

Otherwise I will defer that change for later just to reduce complexity

File: src/text.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+export type VideoTextTracks = {
+  subtitles?: VideoTextTracksProps;
+  captions?: VideoTextTracksProps;
+  descriptions?: VideoTextTracksProps;
+  chapters?: VideoTextTracksProps;
+  metadata?: VideoTextTracksProps;
+};
+
+export type VideoTextTracksProps = {
```
Comment (zzarcon) [inline src/text.tsx:9]: /nit I will just call this `VideoTextTrack`

File: src/text.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  selectedTrackIndex?: number;
+  tracks: Array<VideoTextTrackProps>;
```
Comment (zzarcon) [inline src/text.tsx:11]: /nit `VideoTextTrackProps[]`

File: src/video.tsx (hunk 1 above)
Comment (fredguile) [reply src/video.tsx:12]: My understanding is that `VideoState` would be "read-only" while `VideoActions` is for mutating the state?

Here I'm just looking at reading into `HTMLMediaElement.textTracks?`, the only problem is that that attribute returns a `TextTrackList` that is a bit uneasy for the user: 
- it's a `Iterable` but not an `Array` (calling forEach/filter/map on this is a pain),
- the only method that seems user-friendly is `getTrackById()`, but that imply users control the "id" they pass to it, which is the same "id" property on the <track /> HTML element (wich user don't control)

So basically my idea was to construct this "id" with (kind, lang), and ask users to pass these two params that they own, so that I can return them the proper entry from the `TextTrackList`.

There's a nifty feature of `getTrackById()` that I'm using: if there's several tracks with the same "id" (illegal, but may happen), that function only returns the first entry.

File: README.md (hunk 5 above)
Comment (fredguile) [reply README.md:150]: I think the "wow" effect of our work will be to add a "metadata" track that reference our company resources while playing the video (subtitles/captions are cool, but not so impressive). For this track we need to "build" the feature in our player using `currentActiveCues`. Same for chapters actually...

File: src/video.tsx (hunk 1 above)
Comment (zzarcon) [reply src/video.tsx:12]: Right! 

Yeah, you approach makes sense now. We should use actions only to mutate the state and this is purely for reading
